src/rulesets/oas2/__tests__/oas2-schema.ts

```diff
@@ -1,12 +1,20 @@
 import { RuleType, Spectral } from '../../../spectral';
 import * as ruleset from '../index.json';
+import * as oas2Schema from '../schemas/main.json';
 
 describe('oas2-schema', () => {
   const s = new Spectral();
+
   s.addRules({
-    'oas2-schema': Object.assign(ruleset.rules['oas2-schema'], {
+    'oas2-schema': Object.assign({}, ruleset.rules['oas2-schema'], {
       recommended: true,
       type: RuleType[ruleset.rules['oas2-schema'].type],
+      then: {
+        ...ruleset.rules['oas2-schema'].then,
+        functionOptions: {
```

Comment:
Is this meant to be here? Looks like we’re oassing in the contents not a ref. Is that just to make the unit test work and avoid it being an integration test? If so, should we do the same for oas3?

Reply:
Yeah, it's to avoid it being an integration test.
We don't have any tests for oas3-schema